pkg/curl/schemes.go

```diff
@@ -328,17 +328,16 @@ type SchemeWithRetries struct {
 
 // Fetch implements FileScheme.Fetch for retry wrapper.
 func (s *SchemeWithRetries) Fetch(ctx context.Context, u *url.URL) (io.ReaderAt, error) {
+	var r io.ReaderAt
 	var err error
+
 	s.BackOff.Reset()
 	back := backoff.WithContext(s.BackOff, ctx)
-	for d := time.Duration(0); d != backoff.Stop; d = back.NextBackOff() {
-		if d > 0 {
-			time.Sleep(d)
-		}
-
-		var r io.ReaderAt
-		// Note: err uses the scope outside the for loop.
-		r, err = s.Scheme.Fetch(ctx, u)
+	for d := back.NextBackOff(); d != backoff.Stop; d = back.NextBackOff() {
```

Comment:
should this be < ? I had a problem in a timeout scheme once with != when the timer skipped over the stop value and kept running.

I worry about direct comparison on anything related to time.

Reply:
Stop is a magical value that does not represent time.

https://pkg.go.dev/github.com/cenkalti/backoff/v4#Stop

I think direct comparison is the better option here.